Fugitives - Mid Round Antag

PHCodes opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
Added Fugitives and support for the Mid Round Antags system. 


## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
We didn't have Fugitives yet. 

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
I had to make several changes along the way, some of the largest notes being:
Job is now JobComponent and just takes a string as input. 
TryGetMind outputs the exact mind component we care about. I'm not sure the enclosing query loop even does anything anymore. 
Objectives are now a component that you fetch. This was the biggest change from Nyanotrasen's code. You fetch the Objective and then read out the information that's relevant for marking the entity as an antagonist with the role it should have, etc. I imagine a similar change will need to be made for all mid round antags. 
## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
![image](https://github.com/DeltaV-Station/Delta-v-rebase/assets/47927305/fa99ba1c-f34e-40a4-bcac-49bc3fd80e07)
https://imgur.com/a/69b1uXX
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

<!--
Make sure to take this Changelog template out of the comment block in order for it to show up.
-->
:cl: JJ
- add: Added Fugitives and Midround Antag system.

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurComponent.cs:
```diff
@@ -0,0 +1,30 @@
+        [ViewVariables(VVAccess.ReadWrite)]
+        public EntityPrototype DesiredPrototype = default!;
+    }
+}
```
```suggestion
}

```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```suggestion
}

```

DEATHB4DEFEAT commented on Resources/Locale/en-US/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-thanks-second-5 = My client's pockets are as deep as their world is cold.
+roboisseur-deny-1 = Kindly prepare the correct item.
+roboisseur-deny-2 = This is not what we agreed.
+roboisseur-deny-3 = Not what the client's looking for.
```
```suggestion
roboisseur-deny-3 = Not what the client's looking for.

```

DEATHB4DEFEAT commented on Resources/Prototypes/Nyanotrasen/Entities/Structures/Machines/roboisseur.yml:
```diff
@@ -0,0 +1,24 @@
+      proper: true
+  - type: SpriteFade
+  - type: Tag
+    tags: []
```
```suggestion
    tags: []

```

DEATHB4DEFEAT commented on Resources/Textures/DeltaV/Structures/Machines/roboisseur.rsi/meta.json:
```diff
@@ -0,0 +1,29 @@
+            "name": "roboisseur-impatient2"
+        }
+    ]
+}
```
```suggestion
}

```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+
+namespace Content.Server.Roboisseur.Roboisseur
+{
+    public enum RobossieurVisualLayers : byte
+    {
+        Angry
+    }
```
Odd placement for this, I'd have it near the bottom with a newline or two between it and whatever else is around it

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+        private readonly ISawmill _sawmill = default!;
+
+
+        [ViewVariables(VVAccess.ReadWrite)]
```
What is this doing in a system?

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+            "FoodMothToastedSeeds",
+            "FoodMothCornmealPorridge",
+            "FoodMothBakedCorn",
+            "FoodBreadMoldySlice",
+            "FoodRiceBoiled",
+            "FoodMothEyeballSoup",
+            "FoodMeatRatKebab",
+            "FoodBreadCreamcheese",
+            "FoodSoupOnion",
+            "FoodBurgerAppendix",
+            "FoodBurgerRat",
+            "RegenerativeMesh",
+            "FoodCheeseCurds",
+            "FoodDonkpocketHonkWarm",
+            "FoodOatmeal",
+            "FoodBreadJellySlice",
+            "FoodMothCottonSalad",
+            "FoodBreadMoldy",
+            "FoodDonkpocketSpicyWarm",
+            "FoodCannabisButter",
+            "FoodNoodles",
+            "FoodBreadMeat",
+            "LeavesCannabisDried",
+            "FoodBurgerCheese",
+            "FoodDonkpocketDankWarm",
+            "FoodSpaceshroomCooked",
+            "FoodMealFries",
+            "MedicatedSuture",
+            "FoodDonkpocketWarm",
+            "FoodCakePlain",
+            "DisgustingSweptSoup",
+            "FoodBurgerPlain",
+            "FoodBreadGarlicSlice",
+            "FoodSoupMushroom",
+            "FoodSoupWingFangChu",
+            "FoodBreadMeatXeno",
+            "FoodCakeBrain",
+            "FoodBurgerBrain",
+            "FoodSaladCaesar"
+        };
```
This should probably all be in the component as defaults instead of readonly in a system.

PHCodes replied:
Moved.

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+            SubscribeLocalEvent<RoboisseurComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<RoboisseurComponent, InteractHandEvent>(OnInteractHand);
+            SubscribeLocalEvent<RoboisseurComponent, InteractUsingEvent>(OnInteractUsing);
+        }
```
```suggestion
        }


```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+
+        private void RewardServicer(EntityUid uid, RoboisseurComponent component, int tier)
+        {
+            Random r = new Random();
```
Haven't seen this Type before, I normally use the robust random system dependency.
```suggestion
            var r = new Random();
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+            int rewardToDispense = r.Next(100, 350) + 250 * tier;
+
+            _material.SpawnMultipleFromMaterial(rewardToDispense, "Credit", Transform(uid).Coordinates);
+            if(tier > 1){
```
```suggestion
            if(tier > 1)
            {
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+        private void OnInteractHand(EntityUid uid, RoboisseurComponent component, InteractHandEvent args)
+        {
+            if (!TryComp<ActorComponent>(args.User, out var actor))
+                return;
```
```suggestion
                return;

```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+                return;
+            string message = Loc.GetString(_random.Pick(DemandMessages), ("item", component.DesiredPrototype.Name));
+            if (CheckTier(component.DesiredPrototype.ID) > 1)
+                message = Loc.GetString(_random.Pick(DemandMessagesTier2), ("item", component.DesiredPrototype.Name));
```
```suggestion
                message = Loc.GetString(_random.Pick(DemandMessagesTier2), ("item", component.DesiredPrototype.Name));

```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+
+        private void OnInteractUsing(EntityUid uid, RoboisseurComponent component, InteractUsingEvent args)
+        {
+            if (HasComp<MobStateComponent>(args.Used))
+                return;
+
+            if (!TryComp<MetaDataComponent>(args.Used, out var meta))
+                return;
+
+            if (meta.EntityPrototype == null)
+                return;
```
I don't see a reason for MobStateComponent to be checked here, remove it if there isn't a reason.

`MetaDataComponent` and `TransformComponent` are always on every entity, there is no need for a TryGet or HasComp. Getting these two components can also be done via functions in the system, `MetaData(EntityUid)` and `Transform(EntityUid)` respectively.
```suggestion
            if (HasComp<MobStateComponent>(args.Used) ||
                MetaData(args.Used)?.EntityPrototype == null)
                return;
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+                return;
+
+            var validItem = CheckValidity(meta.EntityPrototype, component.DesiredPrototype);
+
```
```suggestion
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+
+            var nextItem = true;
+
+            if (!validItem)
+            {
+                _chat.TrySendInGameICMessage(uid, Loc.GetString(_random.Pick(RejectMessages)), InGameICChatType.Speak, true);
+                return;
+            }
+            component.Impatient = false;
+            EntityManager.QueueDeleteEntity(args.Used);
+            int tier = CheckTier(component.DesiredPrototype.ID);
+            string message = Loc.GetString(_random.Pick(RewardMessages));
+            if (tier > 1)
+                message = Loc.GetString(_random.Pick(RewardMessagesTier2));
+            _chat.TrySendInGameICMessage(uid, message, InGameICChatType.Speak, true);
+            RewardServicer(args.User, component, tier);
```
```suggestion
            if (!validItem)
            {
                _chat.TrySendInGameICMessage(uid, Loc.GetString(_random.Pick(RejectMessages)), InGameICChatType.Speak, true);
                return;
            }

            component.Impatient = false;
            EntityManager.QueueDeleteEntity(args.Used);

            int tier = CheckTier(component.DesiredPrototype.ID);

            string message = Loc.GetString(_random.Pick(RewardMessages));
            if (tier > 1)
                message = Loc.GetString(_random.Pick(RewardMessagesTier2));
            _chat.TrySendInGameICMessage(uid, message, InGameICChatType.Speak, true);

            RewardServicer(args.User, component, tier);
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+            if (nextItem)
+                NextItem(component);
+        }
+        private bool CheckValidity(EntityPrototype given, EntityPrototype target)
```
```suggestion

        private bool CheckValidity(EntityPrototype given, EntityPrototype target)
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```suggestion

        private int CheckTier(String target)
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```suggestion

            if (_prototypeManager.TryIndex<EntityPrototype>(protoString, out var proto))
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```suggestion
                _sawmill.Error("Roboisseur can't index prototype " + protoString);
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```suggestion

        public List<string> GetAllProtos()
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```suggestion
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```suggestion
            foreach (var recipe in allRecipes)
                allProtos.Add(recipe.Result);

            foreach (var proto in BlacklistedProtos)
                allProtos.Remove(proto);

            return allProtos;
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,355 @@
+        [Dependency] private readonly ChatSystem _chat = default!;
+        [Dependency] private readonly MaterialStorageSystem _material = default!;
+        [Dependency] private readonly AppearanceSystem _appearance = default!;
+        private readonly ISawmill _sawmill = default!;
```
Use the system-wide `Log` function (I think that's what it's called now?)
```suggestion
```

DEATHB4DEFEAT commented on Content.Server/Nyanotrasen/Roboisseur/RoboisseurComponent.cs:
```diff
@@ -0,0 +1,30 @@
+    [RegisterComponent]
+    public sealed class RoboisseurComponent : Component
+    {
+
```
```suggestion
```

DEATHB4DEFEAT commented on Resources/Prototypes/Nyanotrasen/Entities/Structures/Machines/roboisseur.yml:
```diff
@@ -0,0 +1,24 @@
+  - type: Sprite
+    noRot: true
+    drawdepth: Mobs
+    offset: "0.0,0.5"
```
Why's it offset?

```suggestion
    offset: 0.0, 0.5
```

DEATHB4DEFEAT reviewed (changes requested):
What a cool sounding feature 👍

DEATHB4DEFEAT commented:
Should create some media for the PR description.

DEATHB4DEFEAT reviewed (changes requested):
I forgot to check namespaces; move the component, system, locale, and yml to the DeltaV subfolder folder.

leonardo-dabepis commented on Resources/Prototypes/Nyanotrasen/Entities/Structures/Machines/roboisseur.yml:
```diff
@@ -0,0 +1,24 @@
+  parent: BaseStructure
+  id: Roboisseur
+  name: Mr. Butlertron
+  description: It asks for food to deliver to exotic customers across the cosmos. Maybe it's got a Psionic teleporter?
```
the "maybe it's got a psionic teleporter?" seems out of place and doesn't really make sense in this context. Should probably be something like "Powered by the latest technology in bluespace food delivery". At least that's just my opinion 🤪

PHCodes replied:
The idea was to give some bad reason to why it can't be e-magged, at least not yet. Will change to your suggestion since as far as I know bluespace items also cannot be e-magged.

PHCodes replied:
Fixed. I bet you like Midochlorians, too.

DEATHB4DEFEAT commented on Content.Server/DeltaV/NPC/Roboisseur/RoboisseurComponent.cs:
```diff
@@ -0,0 +1,209 @@
+            "FoodMothToastedSeeds",
+            "FoodMothCornmealPorridge",
+            "FoodMothBakedCorn",
+            "FoodBreadMoldySlice",
+            "FoodRiceBoiled",
+            "FoodMothEyeballSoup",
+            "FoodMeatRatKebab",
+            "FoodBreadCreamcheese",
+            "FoodSoupOnion",
+            "FoodBurgerAppendix",
+            "FoodBurgerRat",
+            "RegenerativeMesh",
+            "FoodCheeseCurds",
+            "FoodDonkpocketHonkWarm",
+            "FoodOatmeal",
+            "FoodBreadJellySlice",
+            "FoodMothCottonSalad",
+            "FoodBreadMoldy",
+            "FoodDonkpocketSpicyWarm",
+            "FoodCannabisButter",
+            "FoodNoodles",
+            "FoodBreadMeat",
+            "LeavesCannabisDried",
+            "FoodBurgerCheese",
+            "FoodDonkpocketDankWarm",
+            "FoodSpaceshroomCooked",
+            "FoodMealFries",
+            "MedicatedSuture",
+            "FoodDonkpocketWarm",
+            "FoodCakePlain",
+            "DisgustingSweptSoup",
+            "FoodBurgerPlain",
+            "FoodBreadGarlicSlice",
+            "FoodSoupMushroom",
+            "FoodSoupWingFangChu",
+            "FoodBreadMeatXeno",
+            "FoodCakeBrain",
+            "FoodBurgerBrain",
+            "FoodSaladCaesar"
+        };
```
Needs datafield tags and remove the readonlys

DEATHB4DEFEAT commented on Content.Server/DeltaV/NPC/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,179 @@
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly ChatSystem _chat = default!;
+        [Dependency] private readonly MaterialStorageSystem _material = default!;
+        [Dependency] private readonly AppearanceSystem _appearance = default!;
+        public override void Initialize()
```
```suggestion
        [Dependency] private readonly AppearanceSystem _appearance = default!;


        public override void Initialize()
```

DEATHB4DEFEAT commented on Content.Server/DeltaV/NPC/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,179 @@
+        [Dependency] private readonly AppearanceSystem _appearance = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<RoboisseurComponent, ComponentInit>(OnInit);
```
```suggestion
            base.Initialize();


            SubscribeLocalEvent<RoboisseurComponent, ComponentInit>(OnInit);
```

DEATHB4DEFEAT commented on Content.Server/DeltaV/NPC/Roboisseur/RoboisseurSystem.cs:
```diff
@@ -0,0 +1,179 @@
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+            foreach (var roboisseur in EntityQuery<RoboisseurComponent>())
```
```suggestion
            base.Update(frameTime);


            foreach (var roboisseur in EntityQuery<RoboisseurComponent>())
```

DEATHB4DEFEAT commented on Content.Server/DeltaV/NPC/Roboisseur/RoboisseurSystem.cs:
```suggestion
    }

    public enum RobossieurVisualLayers : byte
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-1 = I humbly request one {$item} for my wealthiest client.
+roboisseur-request-2 = Oodles of dough await you, for a nice {$item}!
+roboisseur-request-3 = There may be an exorbiantly affluent personage in this sector looking for a {$item}.
+roboisseur-request-4 = My more... Adventurous Clients require a {$item}.
```
```suggestion
roboisseur-request-4 = My more... Adventurous Clients require {INDEFINITE($item)} {$item}.
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-1 = I humbly request one {$item} for my wealthiest client.
+roboisseur-request-2 = Oodles of dough await you, for a nice {$item}!
+roboisseur-request-3 = There may be an exorbiantly affluent personage in this sector looking for a {$item}.
```
```suggestion
roboisseur-request-3 = There may be an exorbitantly affluent personage in this sector looking for {INDEFINITE($item)} {$item}.
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-6 = Local changes in this quadrant's stock exchange decteted. Seeking {$item}.
+roboisseur-request-impatient-1 = There is no time left! Bring me {$item}, immediately!
+roboisseur-request-impatient-2 = My market opportunity is dwindling! Quickly, you fools, {$item}!
+roboisseur-request-impatient-3 = Damn you all! Deliver me a {$item} already!
```
```suggestion
roboisseur-request-impatient-3 = Damn you all! Deliver me {INDEFINITE($item)} {$item} already!
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-impatient-2 = My market opportunity is dwindling! Quickly, you fools, {$item}!
+roboisseur-request-impatient-3 = Damn you all! Deliver me a {$item} already!
+roboisseur-request-second-1 = I humbly request one {$item} for one of my snowbound clients.
+roboisseur-request-second-2 = The frozen planet in this sector requests a {$item}!
```
```suggestion
roboisseur-request-second-2 = The frozen planet in this sector requests {INDEFINITE($item)} {$item}!
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-impatient-3 = Damn you all! Deliver me a {$item} already!
+roboisseur-request-second-1 = I humbly request one {$item} for one of my snowbound clients.
+roboisseur-request-second-2 = The frozen planet in this sector requests a {$item}!
+roboisseur-request-second-3 = A local ice cream vendor is looking to trade for a {$item}.
```
```suggestion
roboisseur-request-second-3 = A local ice cream vendor is looking to trade for {INDEFINITE($item)} {$item}.
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-5 = {$item}. It's rare. It's valuable. You can make it, yes?
+roboisseur-request-6 = Local changes in this quadrant's stock exchange decteted. Seeking {$item}.
+roboisseur-request-impatient-1 = There is no time left! Bring me {$item}, immediately!
+roboisseur-request-impatient-2 = My market opportunity is dwindling! Quickly, you fools, {$item}!
```
```suggestion
roboisseur-request-impatient-2 = My market opportunity is dwindling! Quickly, you fools, {THE($item)}!
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-4 = My more... Adventurous Clients require a {$item}.
+roboisseur-request-5 = {$item}. It's rare. It's valuable. You can make it, yes?
+roboisseur-request-6 = Local changes in this quadrant's stock exchange decteted. Seeking {$item}.
+roboisseur-request-impatient-1 = There is no time left! Bring me {$item}, immediately!
```
```suggestion
roboisseur-request-impatient-1 = There is no time left! Bring me {THE($item)}, immediately!
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
```diff
@@ -0,0 +1,25 @@
+roboisseur-request-3 = There may be an exorbiantly affluent personage in this sector looking for a {$item}.
+roboisseur-request-4 = My more... Adventurous Clients require a {$item}.
+roboisseur-request-5 = {$item}. It's rare. It's valuable. You can make it, yes?
+roboisseur-request-6 = Local changes in this quadrant's stock exchange decteted. Seeking {$item}.
```
```suggestion
roboisseur-request-6 = Local changes in this quadrant's stock exchange decteted. Seeking {INDEFINITE($item)} {$item}.
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/roboisseur/roboisseur.ftl:
I didn't test these suggested changes but they should work as intended, make sure for me :)

PHCodes replied:
Testing...

PHCodes replied:
Works.

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/NPC/roboisseur.yml:
```diff
@@ -0,0 +1,24 @@
+  - type: Sprite
+    noRot: true
+    drawdepth: Mobs
+    offset: 0.0, 0.0
```
```suggestion
```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/NPC/roboisseur.yml:
```diff
@@ -0,0 +1,24 @@
+      gender: male
+      proper: true
+  - type: SpriteFade
+  - type: Tag
+    tags: []
```
Is there a tag on the base you are removing or can this component be removed?

DEATHB4DEFEAT reviewed (changes requested):
Grammar is fun.

PHCodes committed:
Metempsychotic machine that definitely works.

PHCodes committed:
Merge branch 'master' into metempsychosis

PHCodes committed:
Cleans up some stuff and gets the system into a demoable state.

Someone else needs to test this, the physics appear to be broken in my local machine.

PHCodes committed:
Merge branch 'master' into metempsychosis

PHCodes committed:
Merge branch 'metempsychosis' of https://github.com/PHCodes/Delta-v-rebase into metempsychosis

PHCodes committed:
Adds Fugitives to the game.

Nyanotrasen's Fugitives added, along with full support for some of the mid round antag systems. Everything seems to work great, very happy with this.

DebugOk reviewed (changes requested):
This should be drafted until metempsychosis is in, instead of adding a bunch of unrelated stuff

PHCodes committed:
Merge branch 'master' into midroundWithMetem

Signed-off-by: JJ <47927305+PHCodes@users.noreply.github.com>

deltanedas commented:
why doesnt this use generic antag + objectives round end text, like ninja

DEATHB4DEFEAT commented on Content.IntegrationTests/Tests/Nyanotrasen/NyanoRandomTests.cs:
```diff
@@ -0,0 +1,119 @@
+            await pairTracker.CleanReturnAsync();
+        }
+    }
+}
```
```suggestion
}

```

DEATHB4DEFEAT commented on Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/humanoid.yml:
```diff
@@ -0,0 +1,11 @@
+      prototypes: [FugitiveStartingGear]
+    - type: Fugitive
+    - type: FugitiveCountdown
+    - type: RandomHumanoidAppearance
```
```suggestion
    - type: RandomHumanoidAppearance

```

DEATHB4DEFEAT commented on Resources/Textures/Nyanotrasen/Structures/Machines/metempsychotic.rsi/meta.json:
```diff
@@ -0,0 +1,27 @@
+    },
+    {
+      "name": "pod_1",
+      "delays": [
+        [
+          0.1,
+          0.1,
+          0.1,
+          0.1,
+          0.1,
+          0.1
+        ]
+      ]
```
```suggestion
      "delays": [[ 0.1, 0.1, 0.1, 0.1, 0.1, 0.1 ]]
```

DEATHB4DEFEAT commented on Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml:
```diff
@@ -0,0 +1,15 @@
+    Oni: 1
+    Arachnid: 1
+    Moth: 1
+    #Harpy: 1 #Still need to be merged. 
```
```suggestion
    Harpy: 1
```

DEATHB4DEFEAT commented on Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml:
```diff
@@ -0,0 +1,15 @@
+    Moth: 1
+    #Harpy: 1 #Still need to be merged. 
+    Reptilian: 0.5
+    #Dwarf: 0.5 #'Dwarves are a no no'
```
```suggestion
    # Dwarf: 0.5 # "Dwarves are a no no"
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/nyanotrasen/station-events/events/fugitive.ftl:
```diff
@@ -0,0 +1,31 @@
+fugitive-user-was-a-fugitive-with-objectives = [color=gray]{$user}[/color] was a fugitive who had the following objectives:
+fugitive-user-was-a-fugitive-with-objectives-named = [color=White]{$name}[/color] ([color=gray]{$user}[/color]) was a fugitive who had the following objectives:
+fugitive-was-a-fugitive-with-objectives-named = [color=White]{$name}[/color] was a fugitive who had the following objectives:
+JobFugitive = Fugitive
```
```suggestion
JobFugitive = Fugitive

```

DEATHB4DEFEAT commented on Resources/Locale/en-US/nyanotrasen/medical/components/metempsychotic.ftl:
```diff
@@ -0,0 +1 @@
+medical-scanner-upgrade-metem = Metempsychotic machine karma bonus
```
```suggestion
medical-scanner-upgrade-metem = Metempsychotic machine karma bonus

```

DEATHB4DEFEAT commented on Resources/Prototypes/Nyanotrasen/Objectives/fugitiveObjective.yml:
```diff
@@ -0,0 +1,12 @@
+  description: You need to escape the station, one way or another. 
+  components:
+  - type: Objective
+    difficulty: 69 #'Make this number absurdly high' - 10/17/2023
```
I don't think dates know how to speak
```suggestion
    difficulty: 69 # "Make this number absurdly high" - 10/17/2023
```

deltanedas replied:
also it wouldnt matter since difficulty is only for tator

DEATHB4DEFEAT commented on Content.Shared/Mind/SharedMindSystem.cs:
```diff
@@ -38,6 +39,16 @@ public override void Initialize()
+
+        mind.ShowExamineInfo = canExamine;
+    }
+    //Nyano - End Code Block.
```
```suggestion
    //Nyano - End Code Block.

```
